src/q7/matrix-INMA2380/exercises/ch1.tex

```diff
@@ -268,16 +263,25 @@ \section{Matrix' algebra}
 
 \exo{2}
 \begin{solution}
-  Asking that they commute to every matrix is actually equivalent to asking
-  that they commute to every element of the base of $\mathcal{R}^{n \times n}$.
-  The canonical base is $e_ie_i^T$ for $i \in \{1, \ldots, n\}$ where
-  $e_k$ is a column vector with 1 at $k$ and 0 elsewhere.
-  We can see that $Ae_ie_i^T = A_{:i}e_i^T$ and $e_ie_i^TA = e_iA_{i:}$.
-
-  If $A_{:i}e_i^T = e_iA_{i:}$, the LHS imposes that only the $i$th column is non-zero
-  and the RHS imposes that only the $i$th line is non-zero.
-  We therefore need $A_{ki} = 0$ for $k \neq i$ and $A_{ik} = 0$ for $k \neq i$.
-  Since this must hold for every $i \in \{1, \ldots, n\}$, $A$ must be diagonal.
+  Let $B$ a square matrix of dimension $n$ commuting with $A = \diag\{a_1,\dots,a_n\}$,
```

Comment:
Why not keeping both solutions ?

Reply:
I first thought the old solution was a copy paste of the previous one but just realised it's not. The question is asking to *prove that a matrix commuting with a diagonal one is itself diagonal*, so we could replace the first sentence by "Asking that it commutes with a diagonal matrix is equivalent to asking that it commutes to every element of the base $e_ie_i^T$ for $i \in \{1, \ldots, n\}$.", and keep both, what do you think?

Reply:
Good idea. "base of $\mathcal{R}^{n \times n}$" is indeed wrong in the old one, it should be diagonal matrices

Reply:
@blegat done in 100ef72